src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -21,9 +23,16 @@ public class OrderServiceImpl implements OrderService {
 
     @Override
     public Order completeOrder(ShoppingCart shoppingCart) {
+        if (shoppingCart == null) {
+            throw new DataProcessingException("Shopping cart is null!");
+        }
         Order order = new Order();
         order.setOrderDate(LocalDateTime.now());
-        order.setTickets(new ArrayList<>(shoppingCart.getTickets()));
+        List<Ticket> ticketList = new ArrayList<>(shoppingCart.getTickets());
```

Comment:
check `shoppingCart` tickets instead, you move extra useless operation and creating extra object each time tickets are empty